### PR TITLE
Refactor execute method of repetition_state.rb

### PR DESCRIPTION
**Pull request by LucianoPC**
Remove if's of method execute on repetition_state

Issue:
Method execute on RepetitionState is really complex #67

### CODE REVIEW

**Comment by rodrigosiqueira** [comment]
We will adopt LGPL instead of GPL, if you agree with this I will proceed with your MR, otherwise I will refuse it. Do you agree with this?

**Comment by LucianoPC** [comment]
Sure

File: lib/kuniri/state_machine/OO_structured_fsm/repetition_state.rb [hunk 1]
```diff
@@ -7,14 +7,20 @@ module OOStructuredFSM
     # Class responsible for handling Repetition state.
     class RepetitionState < OOStructuredState
 
+      MAP_STATE = {
+        StateMachine::METHOD_STATE => "method",
+        StateMachine::CONSTRUCTOR_STATE => "function",
```
Comment (simiaosimis) [inline lib/kuniri/state_machine/OO_structured_fsm/repetition_state.rb:12]: The value of this hashes (lines 12 and 13) are switched, StateMachine::CONSTRUCTOR_STATE should be "constructor" and StateMachine::GLOBAL_FUNCTION_STATE should be "function", you should also make a test that checks this scenario.

File: lib/kuniri/state_machine/OO_structured_fsm/repetition_state.rb [hunk 2]
```diff
@@ -75,6 +60,30 @@ def reset_flag
           @language.flagFunctionBehaviour = StateMachine::NONE_HANDLING_STATE
         end
 
+        def get_capture_lambda(lambdaType)
```
Comment (rodrigosiqueira) [inline lib/kuniri/state_machine/OO_structured_fsm/repetition_state.rb:63]: pLambdaType

File: lib/kuniri/state_machine/OO_structured_fsm/repetition_state.rb [hunk 3]
```diff
@@ -75,6 +60,30 @@ def reset_flag
...
+          capture_lambda = lambda do |has_end_of_block|
+            self.send("#{lambdaType}_capture") if has_end_of_block
+          end
+
+          return capture_lambda
+        end
+
+        def get_list_of_file(pElementFile, elementType)
```
Comment (rodrigosiqueira) [inline lib/kuniri/state_machine/OO_structured_fsm/repetition_state.rb:71]: pElementType

File: lib/kuniri/state_machine/OO_structured_fsm/repetition_state.rb [hunk 4]
```diff
@@ -75,6 +60,30 @@ def reset_flag
...
+          if elementType == MAP_STATE[StateMachine::CONSTRUCTOR_STATE]
+            return pElementFile.global_functions
+          else
+            return pElementFile.classes.last.send("#{elementType}s")
+          end
+        end
+
+        def get_add_repetition_lambda(lambdaType)
```
Comment (rodrigosiqueira) [inline lib/kuniri/state_machine/OO_structured_fsm/repetition_state.rb:79]: pLambdaType

**Comment by rodrigosiqueira** [comment]
I have some suggestions:
1. Take care with the size of line. We use the maximum size of 80 characters.
2. Take care with method sequence. I believe, that will be easier to understand the code if you arrange private method in a linear order. In other words, I suggest this sequence: get_add_repetition_lambda, get_list_of_file, and finally get_captured_lambda.
3. Take care with the complexity in the code, sometimes take time to understand the meaning of the code. Try to put some small comments.
4. Put some extra test. Take a look at simiaosimis comment, it is really relevant.

**Comment by rodrigosiqueira** [comment]
Merged.

**Pull request closed without merging**